src/security/acl_api.c

```diff
@@ -829,3 +873,170 @@ daos_ace_dump(struct daos_ace *ace, uint tabs)
 	print_all_flags(tabs + 1, ace);
 	print_all_perm_types(tabs + 1, ace);
 }
+
+static bool
+type_matches_principal_len(struct daos_ace *ace)
```

Comment:
do you really need all these small functions? they are used once, so maybe some of them are just too small to separate out from the function they are used..

Reply:
My intent in breaking it up was to make very clear in the validate functions what we are actually checking. An easy-to-read checklist. I can condense the helpers back into the parent function, but that does end up requiring comments to fence off individual sections. Which IMO is usually a sign that a helper function would help readability.

Reply:
I think several of them could just be inline.

Reply:
OK. I am inlining some of them.